### PR TITLE
host: Make max job concurrency configurable

### PR DESCRIPTION
**Pull request by josephglanville**
Allow configuring the maximum number of jobs the API will allow to be started concurrently from the command line.

### CODE REVIEW

**Comment by josephglanville** [comment]
@lmars where is it best to raise this for CI?

File: host/host.go [hunk 1]
```diff
@@ -56,6 +56,7 @@ options:
   --peer-ips=IPLIST      join existing cluster using IPs
   --bridge-name=NAME     network bridge name [default: flynnbr0]
   --no-resurrect         disable cluster resurrection
+  --max-job-concurrency  maxiumum number of jobs to start concurrently
```
Comment (titanous) [inline host/host.go]: `maximum`

File: host/http.go [hunk 2]
```diff
@@ -47,6 +47,8 @@ type Host struct {
 
 	listener net.Listener
 
+	maxJobConcurrency uint64
```
Comment (titanous) [inline host/http.go]: Doesn't this need to be an `int`?

Comment (josephglanville) [inline host/http.go]: I thought so too but uint64 works too and is slightly nicer in that it it alleviates the need to further validate for positive input.

**Comment by titanous** [comment]
@josephglanville https://github.com/flynn/flynn/blob/master/test/cluster/cluster.go#L487

File: host/host.go [hunk 3]
```diff
@@ -168,6 +169,12 @@ func runDaemon(args *docopt.Args) {
 		hostID = strings.Replace(hostname, "-", "", -1)
 	}
 
+	var maxJobConcurrency uint64
+	maxJobConcurrency = 4
```
Comment (titanous) [inline host/host.go]:
```go
var maxJobConcurrency uint64 = 4
```

**Comment by titanous** [comment]
LGTM

**Comment by lmars** [comment]
LGTM

**Pull request merged**